src/components/Board.js

```diff
@@ -5,29 +5,171 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
+import UpdateCardForm from './UpdateCardForm';
+import ChangeBoardForm from './ChangeBoardForm';
+// import CARD_DATA from '../data/card-data.json';
 
 class Board extends Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
 
     this.state = {
       cards: [],
+      editing: '',
+      // currentBoard: props.boardName,
+      allBoards: []
     };
   }
 
+  componentDidUpdate(prevProps) {
+    //only update the board if the boardName has changed
+  if (prevProps.boardName !== this.props.boardName) {
+    this.setBoard();
+  }
+}
+
+  componentDidMount() {
+    this.setBoard();
+  }
+
+  setBoard() {
+    const BASE_URL = this.props.url
+    const BOARD = this.props.boardName
+    const CARDS = '/cards'
+
+    axios.get(BASE_URL + BOARD + CARDS)
+    .then((response) => {
+      const cards = response.data;
+
+      this.setState({cards: cards});
+      this.props.updateStatusCallback(``, 'success')
+    })
+    .catch((error) => {
+      this.props.updateStatusCallback(error.message, 'error')
+    })
+
+    axios.get(BASE_URL)
```

Comment:
Why do 2 API calls?

Reply:
Forgot to refactor after implementing the functionality to allow  a user to change the inspiration board.  I was following the whole "make it work, make it right, make it pretty" mantra.  It works, but it is not right or pretty. :*(